### ISSUE TITLE
Select-Control styles break multiple select fields

### ISSUE THREAD

**Issue by MatzeKitt** [WordPress/gutenberg#27166]
**Describe the bug**
The default [select styles](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/select-control/styles/select-control-styles.js) force displaying every select to a regular select field. However, a `select[multiple]` field shouldn’t be forced to have a drop-down arrow nor a maximum height of 30 pixels.

**To reproduce**
1. Create a `SelectControl` with `multiple` attribute.

**Expected behavior**
A `multiple` select field is not styled similar to a regular select field without `multiple` attribute.

**Editor version (please complete the following information):**
- WordPress version: 5.6-RC1-49667
- Tested with and without the Gutenberg plugin
- If the Gutenberg plugin is installed, which version is it? 9.4.1

**Desktop (please complete the following information):**
 - OS: macOS Big Sur
 - Browser: Safari
 - Version: 14.0.1

**Comment by tellthemachines** [comment]
Can confirm `SelectControl` styles are broken when using `multiple`, and issue is visible in [Storybook](https://wordpress.github.io/gutenberg/?path=/story/components-selectcontrol--default) too.

**Comment by ddryo** [comment]
I can reproduce this problem as well.

**Comment by mototeam** [comment]
We have the same issue in WordPress 5.6
Getwid plugin. Screenshot https://prnt.sc/w0i89f
Timetable plugin. Screenshot https://prnt.sc/w0i99q
Is there any plan to fix this in the near future or should we as developers look for a quick fix ourselves?

**Comment by KittMedia** [comment]
As a quick fix, I’m using the following CSS in my block:

```
.impressum-select-multiple.impressum-select-multiple.impressum-select-multiple select[multiple] {
	height: auto;
	padding: 8px;
	
	+ .components-input-control__suffix {
		display: none;
	}
}
```

And I needed to surround my `<SelectControl>` with a `<div class="impressum-select-multiple">`.

**Comment by wistrix** [comment]
I can reproduce this problem too.

**Comment by JohnRDOrazio** [comment]
I started getting this problem too, the component is set to a fixed height of 30px instead of automatically expanding (as a "multiple" select should).
I see the problem is here: https://github.com/WordPress/gutenberg/blob/master/packages/components/src/select-control/styles/select-control-styles.js

**Comment by JohnRDOrazio** [comment]
I also see, reading the component handbook, that the SelectControl is being encouraged for single selections, not for multiple selections. I wonder if this is a simple oversight? 
https://developer.wordpress.org/block-editor/components/select-control/

> SelectControl allow users to select from a single-option menu. 

> Use a select control when: You want users to select a single option from a list.

> Use selects when you have multiple options, and can only choose one.

And yet on the same page it does talk about the possibility of using the `multiple` attribute:

> One important prop to refer is value. If `multiple` is true, value should be an array with the values of the selected options. If `multiple` is false, value should be equal to the value of the selected option.

And towards the bottom of the page, `multiple` is actually listed among the possible attributes:

> multiple # If this property is added, multiple values can be selected. The value passed should be an array.
> – Type: Boolean
> – Required: No

**Comment by JohnRDOrazio** [comment]
I made PR #28811 to fix the wording in the README for the SelectControl component.

**Comment by JohnRDOrazio** [comment]
Since a `SelectControl` component that has the `multiple` property should show a list of options instead of a dropdown, it shouldn't be needing the dropdown arrow, which I believe is defined here:
https://github.com/WordPress/gutenberg/blob/8d1d96d6945c03a5759586b7ab7903e99164765a/packages/components/src/select-control/styles/select-control-styles.js#L82-L97

**Comment by tlewap** [comment]
we have same problem with multiple select field

![image](https://user-images.githubusercontent.com/5683825/109136607-95dc4400-7758-11eb-87e0-dfcd8e051466.png)

**Comment by amirmp3** [comment]
I solved this issue in favor of CSS, check it out:

`
.components-select-control__input[multiple] {
     height: auto !important;
     padding: 0 8px;
}

.components-select-control__input[multiple] + .components-input-control__suffix {
     display: none !important;
}
`
You will have:
![Gutenberg selectcontrol multiple](https://s17.picofile.com/file/8426533842/gutenb.png)


But I think this current style is not user-friendly in my option [Materialize did it great.](https://materializecss.com/select.html)

**Comment by mediaformat** [comment]
There are additional problems with the default styles when using **multiple**

- Longer options are hidden, because of `white-space: pre`
![Screen Shot 2021-04-15 at 1 31 41 PM](https://user-images.githubusercontent.com/2895351/114914374-331a3700-9df0-11eb-9df7-ce501268521a.png)
- Options with text flowing onto a second line, can look confusing, there might be accessibility issues
![Screen Shot 2021-04-15 at 1 39 02 PM](https://user-images.githubusercontent.com/2895351/114915801-0d8e2d00-9df2-11eb-9c37-bbfe50199e18.png)
- Adding some zebra striping seems to help (I also removed the padding-right that made space for the down arrow).
![Screen Shot 2021-04-15 at 1 38 38 PM](https://user-images.githubusercontent.com/2895351/114915888-2e568280-9df2-11eb-9079-ea0d897027c2.png)

So I ended up with this
```
select[multiple] {   
	height: auto;   
	padding-right: 8px;   
	option { 
		white-space: normal; 	
		&:nth-of-type(2n) { 	  
			background-color: #efefef; 	
		}   
	}
 }
```

Shall I attempt a PR?

**Comment by leadclown** [comment]
running into the same issue multiple times. why isn't this fixed yet? why is this not high priority?

currently we can't use multiple selects.

trying combobox instead but please... fix your bugs!

**Comment by mediaformat** [comment]
I was actually trying to find this issue again, so thanks @leadclown for your "contribution" 😜.

> There are additional problems ... Longer options are hidden

Some of my original comments did not consider the [Design Guidelines](https://github.com/WordPress/gutenberg/tree/trunk/packages/components/src/select-control#design-guidelines) for this component

> Menu Items
> 
>     Menu items should be short — ideally, single words — and use sentence capitalization.
>     Do not use full sentences inside menu items.

Otherwise, I'm curious what Design folks think about the default height of the component with `multiple` it seems really small to me.

**Comment by mirka** [comment]
Sorry for the delay, this issue just popped onto my radar.

@jasmussen Do we have any design guidance for this? It's clear that this `multiple` variant isn't used anywhere in Gutenberg, in favor of other UI components (e.g. [FormTokenField](https://wordpress.github.io/gutenberg/?path=/docs/components-formtokenfield--default#dropdown-selector)). Should we deprecate/discourage `multiple` select UIs, and suggest that people use alternatives?

**Comment by jasmussen** [comment]
Tricky one indeed. I can't think of a core reason to have this interface, I always found it fiddly, using modifier keys to add to a selection, and if you misclick you deselect all and have to start over. I don't have strong opinions, and there's likely a great deal of context I'm missing. What can I provide, designwise, to help move things forward?

**Comment by mediaformat** [comment]
@jasmussen You are right, on desktop I prefer the [formtokenfield](https://wordpress.github.io/gutenberg/?path=/story/components-formtokenfield--default&args=__experimentalExpandOnFocus:true) component. On mobile this one does have a more native style UI, compared to formtokenfield

**Comment by mirka** [comment]
I've discussed this with several components folks, and the general direction we converged on was something like:

- When `multiple`, remove the size-dependent height/padding and just reveal the entire select. Basically, we'll just rely on native browser styling.
- In the documentation for the `multiple` prop, we'll recommend alternatives (e.g. FormTokenField, CheckboxControl) which are more in line with the UI patterns we use in Gutenberg.

@mediaformat Are you still available to rebase/tweak your PR (#43213)? If not, I'd be happy to take over 🙂
